datafusion/core/src/physical_optimizer/projection_pushdown.rs

```diff
@@ -1407,29 +1409,29 @@ mod tests {
                 None,
             )),
             Arc::new(CaseExpr::try_new(
-                Some(Arc::new(Column::new("d", 2))),
+                Some(Arc::new(Column::new("d", 3))),
                 vec![
                     (
-                        Arc::new(Column::new("a", 3)) as Arc<dyn PhysicalExpr>,
+                        Arc::new(Column::new("a", 0)) as Arc<dyn PhysicalExpr>,
                         Arc::new(BinaryExpr::new(
-                            Arc::new(Column::new("d", 2)),
+                            Arc::new(Column::new("d", 3)),
                             Operator::Plus,
-                            Arc::new(Column::new("e", 5)),
+                            Arc::new(Column::new("e", 4)),
                         )) as Arc<dyn PhysicalExpr>,
                     ),
                     (
-                        Arc::new(Column::new("a", 3)) as Arc<dyn PhysicalExpr>,
+                        Arc::new(Column::new("a", 0)) as Arc<dyn PhysicalExpr>,
                         Arc::new(BinaryExpr::new(
-                            Arc::new(Column::new("e", 5)),
+                            Arc::new(Column::new("e", 4)),
                             Operator::Plus,
-                            Arc::new(Column::new("d", 2)),
+                            Arc::new(Column::new("d", 3)),
                         )) as Arc<dyn PhysicalExpr>,
                     ),
                 ],
                 Some(Arc::new(BinaryExpr::new(
-                    Arc::new(Column::new("a", 3)),
+                    Arc::new(Column::new("a", 0)),
                     Operator::Modulo,
-                    Arc::new(Column::new("e", 5)),
+                    Arc::new(Column::new("e", 4)),
                 ))),
             )?),
         ];
```

Comment:
Rather than changing this test like that, 
```suggestion
            let projected_exprs: Vec<(Arc<dyn PhysicalExpr>, String)> = vec![
            (Arc::new(Column::new("a", 3)), "a".to_owned()),
            (Arc::new(Column::new("b", 1)), "b_new".to_owned()),
            (Arc::new(Column::new("c", 0)), "c".to_owned()),
            (Arc::new(Column::new("d", 2)), "d_new".to_owned()),
            (Arc::new(Column::new("e", 5)), "e".to_owned()),
            (Arc::new(Column::new("f", 4)), "f_new".to_owned()),
        ];

```
these changes will be simpler and greater coverage

Reply:
Yea, updated.